src/Controller/ModelCollector.php

```diff
@@ -395,8 +447,10 @@ private function searchParentOfInParentedMode(ModelInterface $model)
         }
 
         $condition = $this->relationships->getChildCondition($this->parentProviderName, $this->defaultProviderName);
-        // This is pretty expensive, we fetch all models from the parent provider here.
-        // This can be much faster by using the inverse condition if present.
+        if ([] !== ($inverseFilter = $condition->getInverseFilterFor($model))) {
```

Comment:
Shouldn't this rather be:
```diff
- if ([] !== ($inverseFilter = $condition->getInverseFilterFor($model))) {
+ if ([] !== ($inverseFilter = $condition->getInverseFilterFor($model))) {
```
As I read #541

Reply:
See here https://github.com/contao-community-alliance/dc-general/pull/541/commits/e50422326a7aba8a9eee302b22cff60f60540f96